src/index.js

```diff
@@ -42,49 +38,39 @@ function flatten(node) {
   parts.unshift(name);
 
   return { name, keypath: parts.join(".") };
-}
+};
 
 export default function inject(options) {
   if (!options) throw new Error("Missing options");
 
   const filter = createFilter(options.include, options.exclude);
 
-  let modules;
+  let modules = options.modules;
 
-  if (options.modules) {
-    modules = options.modules;
-  } else {
+  if (!modules) {
     modules = Object.assign({}, options);
     delete modules.include;
     delete modules.exclude;
   }
 
+  const modulesMap = new Map(Object.entries(modules));
+
   // Fix paths on Windows
   if (sep !== "/") {
-    Object.keys(modules).forEach(key => {
-      const module = modules[key];
-
-      modules[key] = Array.isArray(module)
-        ? [module[0].split(sep).join("/"), module[1]]
-        : module.split(sep).join("/");
+    modulesMap.forEach((mod, key) => {
```

Comment:
I believe for..of iteration is preferred as it will be faster on cold runs, i.e. `for (const [mod, key] of modulesMap.entries())` because it skips the jumping in and out of the iteration function.